### PR TITLE
*: Update hasSysDB using check ID instead of Name, and support all other types.

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43818

Problem Summary:

### What is changed and how it works?
* Update `hasSysDB` using check ID instead of Name, and support all other types.
* Clean up the log(get global state and global state change) when the TiDB is notOwner.
* Replace `IsPausedBySystem` with `IsPauseBySystem`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/mock_bootstrap.go [hunk 1]
```diff
@@ -107,7 +108,12 @@ func mockUpgradeToVerLatest(s Session, ver int64) {
 		TestHook.OnBootstrap(s)
 		mustExecute(s, sql)
 		logutil.BgLogger().Info("mock upgrade exec", zap.String("sql", sql))
-		time.Sleep(20 * time.Millisecond)
+		// Make test faster.
+		if intest.InTest {
+			time.Sleep(5 * time.Millisecond)
+		} else {
+			time.Sleep(30 * time.Millisecond)
+		}
```
Comment (tangenta) [inline session/mock_bootstrap.go:116]: I think the whole function `mockUpgradeToVerLatest` is only used in tests?

File: ddl/ddl_worker.go [hunk 2]
```diff
@@ -369,32 +369,41 @@ func (d *ddl) addBatchDDLJobs2Table(tasks []*limitJobTask) error {
 		startTS = txn.StartTS()
 		return nil
 	})
-	if err == nil {
```
Comment (zimulala) [inline ddl/ddl_worker.go:372]: This file only update two place.
One is here, make meet `err != nil` return directly.

File: ddl/ddl_worker.go [hunk 3]
```diff
@@ -369,32 +369,41 @@ func (d *ddl) addBatchDDLJobs2Table(tasks []*limitJobTask) error {
...
+	jobTasks := make([]*model.Job, 0, len(tasks))
+	for i, task := range tasks {
+		job := task.job
+		job.Version = currentVersion
+		job.StartTS = startTS
+		job.ID = ids[i]
+		setJobStateToQueueing(job)
+
+		if d.stateSyncer.IsUpgradingState() {
+			hasSysDB, err := util.HasSysDB(job)
```
Comment (zimulala) [inline ddl/ddl_worker.go:385]: Another is check the `HasSysDB`'s error.

File: session/mock_bootstrap.go (hunk 1 above)
Comment (zimulala) [reply session/mock_bootstrap.go:116]: No，we will use it in IT

Comment (tangenta) [reply session/mock_bootstrap.go:116]: Could you please update the comment to indicate that the new one is for unit test?

File: ddl/ddl_worker.go (hunk 3 above)
Comment (tangenta) [reply ddl/ddl_worker.go:385]: Maybe wrap this block into a separate function `trySetJobToPausing()` or `setJobPausingForUpgrade()`?

File: ddl/util/util.go [hunk 4]
```diff
@@ -325,3 +326,68 @@ func WrapKey2String(key []byte) string {
...
+const mysqlSchemaID = int64(1)
+
+func isSysDB(dbID int64) bool {
+	return dbID == mysqlSchemaID
+}
+
+// HasSysDB checks if it has a system database.
+func HasSysDB(job *model.Job) (bool, error) {
+	// TODO: Handle for the schema ID is 0, like ActionCreatePlacementPolicy.
+	switch job.Type {
```
Comment (tangenta) [inline ddl/util/util.go:339]: I wonder if it's really necessary to enumerate all the possible job types. Can't we determine this by the schema ID or schema name?

**Pull request closed without merging**